crypto/ct/ct_sct.c

```diff
@@ -317,12 +317,11 @@ int SCT_set_source(SCT *sct, sct_source_t source)
 
 int SCT_LIST_set_source(const STACK_OF(SCT) *scts, sct_source_t source)
 {
-    int i, ret = 1;
+    int i, ret = 0;
 
     for (i = 0; i < sk_SCT_num(scts); ++i) {
-        ret = SCT_set_source(sk_SCT_value(scts, i), source);
-        if (ret != 1)
-            break;
+        if (SCT_set_source(sk_SCT_value(scts, i), source))
```

Comment:
Given the inefficiency of making it possible to rollback if an error occurs part way through the list, given the fact that this function can't actually fail part way through at the moment, I'm inclined to remove the function entirely. It's only used in test code anyway, so it's probably not worth having.